integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiJmsRecovery.java

```diff
@@ -0,0 +1,739 @@
+    ExecResult result = assertDoesNotThrow(
+        () -> exec(new String(javacCmd), true));
+    logger.info("javac returned {0}", result.toString());
+    logger.info("javac returned EXIT value {0}", result.exitValue());
+    assertTrue(result.exitValue() == 0, "Client compilation fails");
+  }
+
+  private static Callable<Boolean> runClient(String javaCmd) {
+    return (()  -> {
+      ExecResult result = assertDoesNotThrow(() -> exec(new String(javaCmd), true));
+      logger.info("java returned {0}", result.toString());
+      logger.info("java returned EXIT value {0}", result.exitValue());
+      return ((result.exitValue() == 0));
+    });
+  }
+
+  // Run standalone JMS Client to send/receive message from 
+  // Distributed Destination Member
+  private void runJmsClientOnAdminPod(String action, String queue) {
+
+    String jarLocation = "/u01/oracle/wlserver/server/lib/weblogic.jar";
+    StringBuffer javapCmd = new StringBuffer("kubectl exec -n ");
+    javapCmd.append(domainNamespace);
+    javapCmd.append(" -it ");
+    javapCmd.append(adminServerPodName);
+    javapCmd.append(" -- /bin/bash -c \"");
+    javapCmd.append("java -cp ");
+    javapCmd.append(jarLocation);
+    javapCmd.append(":.");
+    javapCmd.append(" JmsSendReceiveClient ");
+    javapCmd.append(" t3://");
+    javapCmd.append(domainUid);
+    javapCmd.append("-cluster-");
+    javapCmd.append(clusterName);
+    javapCmd.append(":8001 ");
+    javapCmd.append(action);
+    javapCmd.append(" \'");
+    javapCmd.append(queue);
+    javapCmd.append("\'");
+    javapCmd.append(" 100");
+    javapCmd.append(" \"");
+    logger.info("java command to be run {0}", javapCmd.toString());
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Wait for JMS Client to send/recv msg "
+                    + "(elapsed time {0}ms, remaining time {1}ms)",
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(runClient(new String(javapCmd)));
+  }
+
+  private static void createDatabaseSecret(
+        String secretName, String username, String password, 
+        String dburl, String domNamespace) throws ApiException {
+    Map<String, String> secretMap = new HashMap();
+    secretMap.put("username", username);
+    secretMap.put("password", password);
+    secretMap.put("url", dburl);
+    boolean secretCreated = assertDoesNotThrow(() -> createSecret(new V1Secret()
+            .metadata(new V1ObjectMeta()
+                    .name(secretName)
+                    .namespace(domNamespace))
+            .stringData(secretMap)), "Create secret failed with ApiException");
+    assertTrue(secretCreated, String.format("create secret failed for %s in namespace %s", secretName, domNamespace));
+
+  }
+
+  private static void createDomainSecret(String secretName, String username, String password, String domNamespace)
```

Comment:
check CommonMiiTestUtils for this method

Reply:
done